test/utils/wabt_engine.cpp

```diff
@@ -8,17 +8,24 @@
 
 #include <test/utils/adler32.hpp>
 #include <test/utils/wasm_engine.hpp>
+#include <algorithm>
 #include <cassert>
 
 namespace fizzy::test
 {
 class WabtEngine final : public WasmEngine
 {
-    wabt::interp::Environment m_env;
-    wabt::interp::DefinedModule* m_module{nullptr};
-
-    // WABT Executor/Thread with default options.
-    wabt::interp::Executor m_executor{&m_env};
+    // mutable because getting RefPtr from Ref requires non-const Store
+    mutable wabt::interp::Store m_store{([]() constexpr noexcept {
+        wabt::Features features;
+        features.disable_multi_value();
+        features.disable_sat_float_to_int();
+        features.disable_sign_extension();
+        return features;
+    })()};
+    wabt::interp::Module::Ptr m_module;
+    wabt::interp::Instance::Ptr m_instance;
+    wabt::interp::Thread::Ptr m_thread{wabt::interp::Thread::New(m_store, {})};
```

Comment:
Here are the options for `Thread`
https://github.com/WebAssembly/wabt/blob/21350996d5bbcd9cafbaa3a5687e6fb503da060c/src/interp/interp.h#L1025-L1032

Reply:
Default values are fine provided we create it once.

Reply:
So `m_store` does not need to be mutable anymore, right?

Reply:
> So `m_store` does not need to be mutable anymore, right?

It still needs it for `get_memory` where it gets the (smart) pointer to memory, which can be only non-const.